Update for a better Hex release (0.4.1)

yurrriq opened the pull request:
This PR now includes #18 and #19.

---

Use the [LFE Hex package](https://hex.pm/packages/lfe) since it has no dependencies/issues. There is no additional effort involved for users, since `rebar3` has a Hex dependency resolver internally.

Add `pkg_name` so we can keep `'lfe-compile'` without waiting for the Hex team to support (or decide not to support) hyphens:

``` erlang
{pkg_name, rebar3_lfe_compile}
```

~~Bump to [0.4.3](https://hex.pm/packages/rebar3_lfe_compile/0.4.3) to keep versioning sane wrt Hex.~~

yurrriq committed:
Update for better Hex release (0.4.3)

Use the LFE Hex package since it has no dependencies/issues.

Add pkg_name so we can keep 'lfe-compile' without waiting for the Hex
team do support (or decide not to support) hyphens.

Bump to 0.4.3 to keep versioning sane wrt Hex.

yurrriq committed:
Fix typos and add links

yurrriq committed:
Mention both git and hex dep

yurrriq committed:
Add Hex badge (close #16)

yurrriq committed:
Add .travis.yml and badge

yurrriq committed:
Merge branch 'feature/hex-badge' into develop

yurrriq commented:
Note the Travis failures were from me deleting a merged branch before it could try to compile it :smile:

See [this PR's build](https://travis-ci.org/lfe-rebar3/compile/builds/124887890) for an accurate assessment.

oubiwann commented on src/lfe-compile.app.src:
```diff
@@ -1,11 +1,14 @@
 {application, 'lfe-compile',
  [{description, "The LFE rebar3 compiler plugin"},
-  {vsn, "0.4.0"},
+  {vsn, "0.4.3"},
```
The current release is 0.4.0.

Let's change this to 0.4.1.

oubiwann commented on rebar.config:
```diff
@@ -1,5 +1,5 @@
 {erl_opts, []}.
```
I don't want to use hex by default, so let's revert the change to this file.

oubiwann commented on rebar.lock:
```diff
@@ -1,4 +1 @@
-  {git,"https://github.com/rvirding/lfe.git",
-       {ref,"cd96efc3d2a73ecff7f3e2ef3563b73b098e4844"}},
-  0}].
+[{<<"lfe">>,{pkg,<<"lfe">>,<<"1.0.2">>},0}].
```
I don't want to use hex by default, so let's revert the change to this file.

Same thing for `rebar.config`.

yurrriq replied:
We can't do this. If you don't want to use Hex, then we can't at all, unless we do some profile voodoo or something. If you're opposed to Hex support, I guess I could maintain a fork/branch that's identical except for the lfe dep spec.

oubiwann committed:
Split examples into two.

oubiwann committed:
Update lfe-compile.app.src

oubiwann committed:
Revert to non-hex rebar.config.

oubiwann committed:
Revert to non-hex rebar.lock.

oubiwann commented:
Huh, it looks like Github now supports in-PR edits ... so I addressed my own in-line comments and updated files.

oubiwann commented:
What do you mean "keep versioning sane with hex.pm"?

Why do we need to jump version numbers?

yurrriq commented:
Well Hex has 0.4.1 0.4.2 and 0.4.3. I'm not sure if it will treat 0.4.0 as the latest since it will be the latest chronologically or as an old version because of the numbers.

yurrriq commented on rebar.lock:
```diff
@@ -1 +1,4 @@
-[{<<"lfe">>,{pkg,<<"lfe">>,<<"1.0.2">>},0}].
+[{<<"lfe">>,
```
We can't do this. Hex packages must have all Hex packages as dependencies. There's also no reason not to us the LFE hex package, as it has no dependencies and involves no additional effort by the user beyond installing `rebar3`.

yurrriq commented:
Some of your updates break this PR..

yurrriq commented on rebar.config:
```diff
@@ -1,5 +1,5 @@
 
 {deps, [
-  {lfe, "1.0.2"}
+  {lfe, {git, "https://github.com/rvirding/lfe.git", {tag, "v1.0.2"}}}
```
We can't do this. Hex packages must have all Hex packages as dependencies. There's also no reason not to use the [LFE hex package](https://hex.pm/packages/lfe), as it has no dependencies and involves no additional effort by the user beyond installing `rebar3`.

yurrriq commented on README.md:
````diff
@@ -50,18 +50,22 @@ $ rebar3 compile
-   %% not both!
+```erlang
+{plugins, [
+   {lfe-compile, "0.4.1", {pkg, rebar3_lfe_compile}}
````
:+1: to splitting up the examples, but this introduces a typo.

yurrriq committed:
Revert "Revert to non-hex rebar.config."

This reverts commit 94719f3404253af3f325e65c61b983d72224e7d9.

yurrriq committed:
Revert "Revert to non-hex rebar.lock."

This reverts commit 987dc82076a3c959eef8c28641d21688de22007f.

yurrriq committed:
Fix typo

Fix typo from 3341151ae36f04439b751bcf8777f5ee49794c23:

    s/lfe-config/'lfe-config'/

yurrriq commented:
Re: versioning, I've reverted the other ones so your version is fine.

yurrriq commented:
Ok, I've [tagged](https://github.com/lfe-rebar3/compile/releases/tag/0.4.1) and [published](https://hex.pm/packages/rebar3_lfe_compile) [0.4.1](https://github.com/lfe-rebar3/compile/pull/17/commits/fe8808e6bafa3783ca47e755a709818c0b67702a).
